tools/generators/migrate-v8-pkg/index.ts

```diff
@@ -33,18 +56,131 @@ function normalizeOptions(tree: Tree, options: MigrateV8PkgGeneratorSchema) {
 }
 
 /**
- * NOTE: remove this if your generator doesn't process any static/dynamic templates
+ *
+ * Narrows down Schema definition to true runtime shape after {@link validateSchema} is executed
+ * - also properly checks of truthiness of provided value
  */
-function addFiles(tree: Tree, options: NormalizedSchema) {
-  const templateOptions = {
-    ...options,
-    tmpl: '',
+function hasSchemaFlag<T extends MigrateV8PkgGeneratorSchema, K extends keyof T>(
+  schema: T,
+  flag: K,
+): schema is T & Record<K, NonNullable<T[K]>> {
+  return Boolean(schema[flag]);
+}
+async function validateSchema(tree: Tree, schema: MigrateV8PkgGeneratorSchema) {
+  const newSchema = { ...schema };
+
+  if (newSchema.name && newSchema.stats) {
+    throw new Error('--name and --stats are mutually exclusive');
+  }
+
+  const shouldValidateNameInput = () => {
+    return !newSchema.name && !newSchema.stats;
   };
 
-  generateFiles(
-    tree,
-    path.join(__dirname, 'files'),
-    path.join(options.projectConfig.root, options.name),
-    templateOptions,
+  if (shouldValidateNameInput()) {
+    throw new Error(`--name cannot be empty. Please provide name of the package.`);
+  }
+
+  if (newSchema.name) {
+    const projectNames = newSchema.name.split(',').filter(Boolean);
```

Comment:
Given we support a comma separated list of packages, we should change the schema docs to reflect this.

Reply:
oh yeah that should be done for v9 as well, somehow I thought I did that, shame ! :D

Reply:
ah ok v9 is all good 
![image](https://user-images.githubusercontent.com/1223799/164287395-25c0a26e-7803-41ce-9c2d-04b51af8accb.png)

Reply:
aslo note ATM this doesn;t do anything. the imporant part is `--stats` flag